library/std/src/path/tests.rs

```diff
@@ -7,6 +7,7 @@ use crate::rc::Rc;
 use crate::sync::Arc;
 use core::hint::black_box;
 
+#[allow(unused_macro_rules)]
```

Comment:
I think you need to allow unknown lints here, like this:

```suggestion
#[allow(unknown_lints, unused_macro_rules)]
```